src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,83 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationService registrationService;
+    private User actual;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        actual = new User();
+    }
+
+    @Test
+    void register_validData_Ok() {
+        actual.setAge(20);
+        actual.setLogin("validLogin");
+        actual.setPassword("validPass");
+        User registered = registrationService.register(actual);
+        assertEquals(actual.getLogin(), registered.getLogin());
+        assertEquals(actual.getPassword(), registered.getPassword());
+        assertEquals(actual.getAge(), registered.getAge());
+    }
+
+    @Test
+    void register_ageUnder18_notOk() {
+        actual.setAge(16);
+        actual.setLogin("youngUser");
+        actual.setPassword("password");
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(actual));
+    }
+
+    @Test
+    void register_loginAlreadyExistsInStorage_notOk() {
+        actual.setAge(20);
+        actual.setLogin("duplicateLogin");
+        actual.setPassword("password");
+        registrationService.register(actual);
```

This test seeds storage by calling the method under test: `registrationService.register(actual);`. The checklist explicitly requires: "Use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` method to add users for testing." Replace this seeding with low-level storage insertion so tests don't rely on the method under test to prepare data (and so a failing register won't silently hide a test's setup failure).